File: 2021/09/04/MaxConsecutiveOnes.java
/**
 * https://leetcode.com/problems/max-consecutive-ones/
 * @author Andrew Jarombek
 * @since 9/4/2021
 */

public class MaxConsecutiveOnes {
    public static int findMaxConsecutiveOnes(int[] nums) {
        int max = 0;
        int lastZero = -1;

        for (int i = 0; i < nums.length; i++) {
            if (nums[i] == 0) {
                max = Math.max(max, i - lastZero - 1);
                lastZero = i;
            }
        }

        max = Math.max(max, nums.length - lastZero - 1);

        return max;
    }

    public static void main(String[] args) {
        assert findMaxConsecutiveOnes(new int[] {1, 1, 0, 1, 1, 1}) == 3;
    }
}

File: 2022/09/06/maxConsecutiveOnes.js
/**
 * https://leetcode.com/problems/max-consecutive-ones/
 * @author Andrew Jarombek
 * @since 9/6/2022
 */

const findMaxConsecutiveOnes = function(nums) {
    let result = 0;
    let current = 0;

    for (let num of nums) {
        current = num ? current + 1 : 0;
        result = Math.max(result, current);
    }

    return result;
};

let result = findMaxConsecutiveOnes([1,1,0,1,1,1]);
console.assert(result === 3);

result = findMaxConsecutiveOnes([1,0,1,1,0,1]);
console.assert(result === 2);

result = findMaxConsecutiveOnes([1,0,1,0,0,1]);
console.assert(result === 1);
